library/net.cpp

```diff
@@ -1100,6 +1100,8 @@ static int dnet_socket_connect(dnet_node *node, dnet_addr_socket_set &original_l
 	}
 
 	pthread_mutex_lock(&state->lock);
+
+	assert(state->succeed_count + state->failed_count <= state->total_count); // 'less' could be on timeout
```

Comment:
You can not just crash the server because of some issues. Just can not. This is not allowed. If coredumps are turned off this ends up with spontaneous restarts which can not be debugged at all.

You have to write error message and mitigate error - either to reset connection or return error to client or whatever else, but not simply kill the server.